framework/src/utils/MultiMooseEnum.C

```diff
@@ -195,37 +195,29 @@ MultiMooseEnum::assign(InputIterator first, InputIterator last, bool append)
   if (!append)
     clear();
 
-  std::copy(first, last, std::back_inserter(_current_names_preserved));
   for (InputIterator it = first; it != last; ++it)
   {
-    std::string upper(*it);
-    std::transform(upper.begin(), upper.end(), upper.begin(), ::toupper);
-
+    std::string upper(MooseUtils::toUpper(*it));
     checkDeprecatedBase(upper);
+    std::set<MooseEnumItem>::const_iterator iter = find(upper);
```

Comment:
Recommend catching these complex types with `auto`. If we change them later, we'll only have to change them in one place.